---
title: C6312 | Dokumentace Microsoftu
ms.date: 11/15/2016
ms.prod: visual-studio-dev14
ms.technology: vs-ide-code-analysis
ms.topic: reference
f1_keywords:
- C6312
helpviewer_keywords:
- C6312
ms.assetid: 1fc8b9a1-e6ba-4799-84c3-31f289576cca
caps.latest.revision: 15
author: mikeblome
ms.author: mblome
manager: jillfra
ms.openlocfilehash: 5011bd00f9fa61b646419b80afcd86dca686a005
ms.sourcegitcommit: 8b538eea125241e9d6d8b7297b72a66faa9a4a47
ms.translationtype: MT
ms.contentlocale: cs-CZ
ms.lasthandoff: 01/23/2019
ms.locfileid: "54752630"
---
# <a name="c6312"></a>C6312
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

upozornění C6312: Možná nekonečná smyčka: použití konstanty EXCEPTION_CONTINUE_EXECUTION ve výrazu filtru výjimky Try-except  
  
 Toto upozornění znamená použití konstanty `EXCEPTION_CONTINUE_EXECUTION` (nebo jiné konstanty, který se vyhodnotí na hodnotu -1) ve výrazu filtru strukturovanou obslužnou rutinou. Použití konstantní hodnoty `EXCEPTION_CONTINUE_EXECUTION` může vést k nekonečné smyčce. Například pokud došlo k výjimce hardware, se restartuje instrukce, který způsobil výjimku. Pokud je stále chybný adresu, která způsobila výjimku, další výjimku dojde k a zpracovat stejným způsobem. To způsobí, že nekonečnou smyčku.  
  
 Explicitní volání konstruktoru `RaiseException` bude přímo příčina nekonečnou smyčku, ale bude pokračovat provádění kódu v chráněném bloku. To může neočekávané a může vést k nekonečné smyčky, pokud `RaiseException` byl použit k přesměrování neplatný ukazatel.  
  
 Obvykle `EXCEPTION_CONTINUE_EXECUTION` má být vrácen pouze ve funkci s názvem ve výrazu filtru, který má možnost opravit buď ukazatel, který způsobil výjimku nebo že základní paměť.  
  
## <a name="example"></a>Příklad  
 Následující kód vygeneruje toto upozornění:  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
void f (char *ptr)  
{  
  __try   
  {  
    // exception occurs if the caller passes null ptr  
    // code...     
   *ptr = '\0';  
  }   
  __except (EXCEPTION_CONTINUE_EXECUTION)  
    // When EXCEPTION_CONTINUE_EXECUTION is used, the handler  
    //  block of the structured exception handler is not executed.   
  {  
    puts("This block is never executed");  
  }  
}  
```  
  
 Chcete-li opravit toto upozornění, použijte následující kód:  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
void f (char *ptr)  
{  
  __try   
  {  
    // exception occurs if the caller passes null ptr  
    // code...  
    *ptr = '\0';  
  }   
  __except (GetExceptionCode()== EXCEPTION_ACCESS_VIOLATION ?   
               EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH)   
  {  
    puts("Error Occurred");  
  }  
}  
```
